.github/workflows/go.yml

```diff
@@ -0,0 +1,32 @@
+name: Go
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: ["1.18", "1.21"]
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: ${{ matrix.version }}
+        
+    - name: Build
+      run: go build -v ./...
+      
+    - name: Test
+      run: go test -v ./...
```

Comment:
Will this test sub modules?

Reply:
It will now: 2f12416dfacd97bd102b61a417a2b748a37ce5db